### DISCUSSION TITLE
Support for Lexus

### DISCUSSION THREAD

**Discussion by Minglarn** [DurgNomis-drol/ha_toyota#32] [Ideas]
Hi! 
Finaly a integration for the Toyota... 
Just one question....
Is it possible to make one for Lexus cars aswell?

Toyota and Lexus are using the same platform as I know...

**Comment by DurgNomis-drol** [comment]
This is out of scope for this integration. As there is no documentation on the the Lexus Link API (there where none for Toyota), someone would have to reverse engineer it. It is very difficult to do this if you don't own a vehicle to do it with, as this often requires a lot of trial and error. Only if it is very very similar or identical to the Toyota API will I consider to including it here. 

If you want to take on the task, I can give you some guidance and help. 😊

**Reply by Minglarn** [reply]
Would bee glad to help... 
But my knowledge about programing is low, specially python... PHP was my main language couple of years ago... :D

**Reply by DurgNomis-drol** [reply]
Python is very easy! The difficult part is to reverse engineer the endpoints. 

I use Chrome developer tools for this, you can access this directly in Chrome. Maybe you know this from your days with PHP 😄. Chrome developer tools runs in a split view with your webpage, in this case the Lexus link webportal. I then use the network tap to see all the request while I login, access the different services and so forth.

Another way is to decompile the Android App, though i have never done something like that, but as far as i know it should be quite trivial to do so and then read all the endpoints through the code.

You can look at how i build the underlying package for this integration to get an idea of how to build a python library. [mytoyota](https://github.com/DurgNomis-drol/mytoyota) Though i'm not a professional developer, so this could probably be done in a better and more efficient way. The code have changed a lot over the last 4 month, as I have gotten better at coding. I started coding python in January.

Feel free to ask more questions if you have some. 😄

**Reply by Minglarn** [reply]
I just started to check what the dev mode spits out... 
I don't have the time do this, but i could squeeze this as a nightly project when the family sleeps. 😄

**Comment by DurgNomis-drol** [comment]
UPDATE

I have seen indication that the API may be same. If I change this header "X-TME-BRAND: TOYOTA" to  "X-TME-BRAND: INFINITI" i actually get an result back, which makes me think that the API may be the same for some things. Though i would still need access to car to do it.

**Reply by Minglarn** [reply]
Alright!!
Maybe I could assist you with some info if it's required....

**Reply by DurgNomis-drol** [reply]
Answering both this and the other above:

That would be awesome 😄 

Another way is to decompile the Lexus link app for android, i just did this for the Toyota app to find out more, and it was relatively easy.

**Reply by Minglarn** [reply]
Nice! 
If needed, you can use my Lexus (Europe) account... IF you promise to not do nasty stuff with it 🤣

**Reply by DurgNomis-drol** [reply]
Thanks for the offer, I hope it won't be necessary. I really hope that most of the endpoints, if not all, are identical. 😃

**Reply by Minglarn** [reply]
Let's hope so... 👍

**Comment by zgadson** [comment]
I just asked about this on the Home Assistant forums as I just picked up a 2018 Lexus IS and I'm anxious to integrate into my dashboard. What can I do to help!?

**Comment by zgadson** [comment]
Any updates on this? Definitely interested, and would be happy to use my car as a guinea pig!

I have quite a few connectivity options as well. My [Grom VLine2](https://gromaudio.com/store/vline/lexus-is250-350-f-2017-2020-vline-carplay-android-auto-infotainment-system-navigation-upgrade.html) is Android OS based to provide CarPlay & Android Auto with a cellular connection, and a [VeePeak Bluetooth OBD Tool](https://www.veepeak.com/product/obdcheck-ble/).

Please let me know what I can do to help!

**Reply by DurgNomis-drol** [reply]
No unfortunately not.

In the mean time you can probably use this integration for your ODB2 tool https://www.home-assistant.io/integrations/torque/
